lib/cext/preprocess.rb

```diff
@@ -26,10 +26,18 @@
 		case Qfalse_int_const : mybool = 0; break;
 EOF
 
-PATCHED_FILES = {'xml_node_set.c'      => {:match =>  /[[:blank:]]*?switch\s*?\(.*?Qnil:/m, 
-                                           :replacement => XML_NODE_SET_PATCH},
-                 'pg_binary_encoder.c' => {:match => /[[:blank:]]*?switch\s*?\(.*?Qfalse\s*?:.*?break;/m,
-                                           :replacement => PG_BINARY_ENCODER_PATCH}}
+PATCHED_FILES = {'xml_node_set.c'      => [{:match =>  /[[:blank:]]*?switch\s*?\(.*?Qnil:/m, 
+                                           :replacement => XML_NODE_SET_PATCH}],
+                 'pg_binary_encoder.c' => [{:match => /[[:blank:]]*?switch\s*?\(.*?Qfalse\s*?:.*?break;/m,
+                                           :replacement => PG_BINARY_ENCODER_PATCH}],
+                 'bytebuffer.c'        => [{:match => /(static VALUE .*?) = Qnil;/,
+                                           :replacement => '\1;'}],
+                 'monitor.c'           => [{:match => /(static VALUE .*?) = Qnil;/,
```

Comment:
It would be great if we could include the gem name in the key here.
Otherwise generic file names like `bytebuffer.c` etc might be present in many gems and unintentionally be patched.